autorest/codegen/serializers/__init__.py

```diff
@@ -74,6 +84,50 @@ def serialize(self) -> None:
         if self.code_model.options["models_mode"] and (self.code_model.schemas or self.code_model.enums):
             self._serialize_and_write_models_folder(env=env, namespace_path=namespace_path)
 
+        if self.code_model.options["package_mode"]:
+            self._serialize_and_write_package_files(out_path=namespace_path)
+
+
+    def _serialize_and_write_package_files(self, out_path: Path) -> None:
+        def _serialize_and_write_package_files_proc(**kwargs: Any):
+            for template_name in package_files:
+                output_name = out_path / (template_name.replace(".jinja2", ""))
+                if not output_name.exists():
```

Comment:
I think it might be better to use the `read_file` check that we have in . I know looking at the path by itself can have weird behavior when we're using downloaded autorest and depending how users specify output folder, so it's better to let autorest handle the output path stuff.

Reply:
Also, while I think most of the packaging files should not be regenerated if they already exist, I feel it would be worth it to regenerate with setup.py, because setup.py contains version information. cc @lmazuel for his thoughts

Reply:
(1) let us use `read_file` instead of `exists()`
(2) will override `setup.py` each time
(3) as long as CHANGELOG.md (case-insensitive) exist, we will never override it.